pkg/controller/actuator_reconcile.go

```diff
@@ -102,6 +103,14 @@ func (a *actuator) Reconcile(ctx context.Context, network *extensionsv1alpha1.Ne
 		}
 	}
 
+	if cluster.Shoot.Spec.Networking.Nodes != nil && len(*cluster.Shoot.Spec.Networking.Nodes) > 0 {
+		autodetectionMode := fmt.Sprintf("cidr=%s", *cluster.Shoot.Spec.Networking.Nodes)
+		if networkConfig == nil {
+			networkConfig = &calicov1alpha1.NetworkConfig{IPv4: &calicov1alpha1.IPv4{}}
+		}
+		networkConfig.IPv4.AutoDetectionMethod = &autodetectionMode
```

Comment:
Would the following not be more concise:
networkConfig.IPv4.AutoDetectionMethod = "cidr=" + *cluster.Shoot.Spec.Networking.Nodes

If yes, we can get rid of line 19 and 107.

Reply:
we need a pointer here right

Reply:
Ok, but the string concatenation should be possible without fmt, right?

Reply:
Sure, thats possible. Over the extensions we used `Sprintf` function but I have no strong opinion here.